lib/place_v7.ml

```diff
@@ -168,28 +168,17 @@ let places_to_string inverse pl =
 exception List_too_long
 
 let get_opt conf =
-  let add_birth = p_getenv conf.env "bi" = Some "on" in
-  let add_baptism = p_getenv conf.env "ba" = Some "on" in
-  let add_death = p_getenv conf.env "de" = Some "on" in
-  let add_burial = p_getenv conf.env "bu" = Some "on" in
-  let add_marriage = p_getenv conf.env "ma" = Some "on" in
-  let f_sort = p_getenv conf.env "f_sort" = Some "on" in
-  let up = p_getenv conf.env "up" = Some "on" in
-  let a_sort = p_getenv conf.env "a_sort" = Some "on" in
-  let lower = p_getenv conf.env "lower" = Some "on" in
-  let word = p_getenv conf.env "word" = Some "on" in
-  let any = p_getenv conf.env "any" = Some "on" in
-  (if add_birth then "&bi=on" else "")
-  ^ (if add_baptism then "&ba=on" else "")
-  ^ (if add_death then "&de=on" else "")
-  ^ (if add_burial then "&bu=on" else "")
-  ^ (if add_marriage then "&ma=on" else "")
-  ^ (if f_sort then "&f_sort=on" else "")
-  ^ (if up then "&up=on" else "")
-  ^ (if a_sort then "&a_sort=on" else "")
-  ^ (if lower then "&lower=on" else "")
-  ^ (if word then "&word=on" else "")
-  ^ if any then "&any=on" else ""
+  (if p_getenv conf.env "bi" = Some "on" then "&bi=on" else "")
+  ^ (if p_getenv conf.env "ba" = Some "on" then "&ba=on" else "")
+  ^ (if p_getenv conf.env "de" = Some "on" then "&de=on" else "")
+  ^ (if p_getenv conf.env "bu" = Some "on" then "&bu=on" else "")
+  ^ (if p_getenv conf.env "ma" = Some "on" then "&ma=on" else "")
+  ^ (if p_getenv conf.env "f_sort" = Some "on" then "&f_sort=on" else "")
+  ^ (if p_getenv conf.env "up" = Some "on" then "&up=on" else "")
+  ^ (if p_getenv conf.env "a_sort" = Some "on" then "&a_sort=on" else "")
+  ^ (if p_getenv conf.env "lower" = Some "on" then "&lower=on" else "")
+  ^ (if p_getenv conf.env "word" = Some "on" then "&word=on" else "")
+  ^ if p_getenv conf.env "any" = Some "on" then "&any=on" else ""
```

Comment:
Nice.
Maybe we can factorize those lines a little more by doing something like:

```ocaml
let get_opt conf =
   let to_url_param s =
      if p_getenv conf.env s = Some "on" then (Printf.sprintf "&%s=on" s) else "")
   in
  let l = List.map to_url_param [ "bi"; "ba"; "de";"bu";"ma";"f_sort";"up";"a_sort";"lower";"word";"any"] in
  String.concat "" l
```

Reply:
Thanks